index.js

```diff
@@ -76,29 +76,36 @@ const camelCase = (input, options) => {
 		return '';
 	}
 
+	const toLowerCase = options.locale === false ?
+		string => string.toLowerCase() :
+		string => string.toLocaleLowerCase(options.locale);
+	const toUpperCase = options.locale === false ?
+		string => string.toUpperCase() :
+		string => string.toLocaleUpperCase(options.locale);
+
 	if (input.length === 1) {
-		return options.pascalCase ? input.toLocaleUpperCase(options.locale) : input.toLocaleLowerCase(options.locale);
+		return options.pascalCase ? toUpperCase(input) : toLowerCase(input);
 	}
 
-	const hasUpperCase = input !== input.toLocaleLowerCase(options.locale);
+	const hasUpperCase = input !== toLowerCase(input);
 
 	if (hasUpperCase) {
-		input = preserveCamelCase(input, options.locale);
+		input = preserveCamelCase(input, toLowerCase, toUpperCase);
 	}
 
 	input = input.replace(LEADING_SEPARATORS, '');
 
 	if (options.preserveConsecutiveUppercase) {
-		input = preserveConsecutiveUppercase(input);
+		input = preserveConsecutiveUppercase(input, toLowerCase);
 	} else {
-		input = input.toLocaleLowerCase();
+		input = toLowerCase(input);
```

Comment:
I think this fixes a bug "by accident" where the locale hasn't been applied previously.